[Doc] Move CONTRIBUTING to docs site

russellb opened the pull request:
The `CONTRIBUTING.md` file has some good information for contributors.
It would be nice to have a place to organize additional documentation
aimed at contributors, so move this content over to the docs site to
seed a new `docs/source/contributing` directory.

Replace `CONTRIBUTING.md` with a link to the new home for this
content.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

russellb committed:
[Doc] Move CONTRIBUTING to docs site

The `CONTRIBUTING.md` file has some good information for contributors.
It would be nice to have a place to organize additional documentation
aimed at contributors, so move this content over to the docs site to
seed a new `docs/source/contributing` directory.

Replace `CONTRIBUTING.md` with a link to the new home for this
content.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

youkaichao commented on CONTRIBUTING.md:
maybe we can directly remove this file?

russellb replied:
I thought about it, but it's really common to have this file, so keeping it as a pointer seemed harmless enough to me

simon-mo commented on docs/source/index.rst:
```diff
@@ -150,6 +150,12 @@ Documentation
+   :maxdepth: 2
+   :caption: Contributing
+
+   contributing/overview.md
```
Suggest moving this to first one inside Developer Documentation

russellb replied:
@simon-mo I wasn't really sure what the desired scope of that section was. It mostly looks like Python API documentation with a target audience of people using vllm as a library.

and the `Profiling vLLM` page that does seem more aligned to the "contributing" section I was proposing here.

and then there's the `Dockerfile` page that I'm not sure really fits in either one.

in summary, we could have

* API Documentation -- targeted at people using vllm as a library
* Contributing -- targeted at people working on the vLLM source

and I wouldn't use "Developer Documentation" in either place, since that's a bit ambiguous between the two personas.

Finally, the last alternative I see would be to keep contributor information out of the docs site completely and keep it in its own directory to be browsed on github. That's another way to keep separation between the personas.

russellb replied:
I just pushed another commit with the changes I proposed above.

* move profiling and dockerfile content to the new "Contributing" section
* rename "Developer Documentation" to "API Documentation"

DarkLight1337 replied:
I think we should also move the "vLLM Paged Attention" page as well as the "Input Processing Pipeline" and "Adding a Multimodal Plugin" pages to "Contributing".

russellb replied:
Thanks for highlighting those docs. I think this warrants a third section:

1. `API Documentation` - API reference documents, primarily aimed at those using vLLM as a library
2. `Design` - design docs, discussing how parts of vllm work (more technical)
3. `Contributing` - information on how to work with vllm as a developer and contributor (more policy and process oriented)

russellb replied:
> Thanks for highlighting those docs. I think this warrants a third section:
> 
> 1. `API Documentation` - API reference documents, primarily aimed at those using vLLM as a library
> 2. `Design` - design docs, discussing how parts of vllm work (more technical)
> 3. `Contributing` - information on how to work with vllm as a developer and contributor (more policy and process oriented)

just finished organizing into these sections

russellb committed:
[Doc] Rename dev docs section

Rename the "Developer Documentation" section to "API Documentation"
since that's the majority of the content in this section. Move the
"dockerfile" and "profiling" content over to the new "Contributing"
section, since it's more aligned to that.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

DarkLight1337 commented on docs/source/contributing/overview.md:
```diff
@@ -0,0 +1,60 @@
+
+If you encounter a bug or have a feature request, please [search existing issues](https://github.com/vllm-project/vllm/issues?q=is%3Aissue) first to see if it has already been reported. If not, please [file a new issue](https://github.com/vllm-project/vllm/issues/new/choose), providing as much relevant information as possible.
+
+> [!IMPORTANT]
```
This doesn't to be properly formatted as an important block in the generated docs.

russellb replied:
Ah, good catch. This is github-flavored-markdown specific

russellb replied:
I fixed this by converting the doc to `rst` format, which supports this.

russellb committed:
[Doc] Convert contributing doc to rst

Conver the old CONTRIBUTING.md file content to rst format, instead.
This will allow the `note` and `important` admonition blocks to be
rendered properly. The content previously used github flavored
markdown specific syntax for an `important` message.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

russellb committed:
[Doc] moved paged attention doc to contributing section

This doc is aimed more at people working on vllm itself, so move it
into the contributing section.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

russellb committed:
[Doc] Create a new Design section

Signed-off-by: Russell Bryant <rbryant@redhat.com>

DarkLight1337 reviewed (approved):
LGTM now, thanks for reorganizing the docs!

Some future improvements to think about:

- Group the pages related to Installation and Deployment to avoid taking up so much space in the sidebar
- Reorganize the pages under Models (why are Engine Arguments / Performance and Tuning placed under there?)
